nominal/experimental/stream_v2/_write_stream.py

```diff
@@ -116,9 +145,19 @@ def enqueue_from_dict(
             timestamp: The common timestamp to use for all enqueued items.
             channel_values: A dictionary mapping channel names to their values.
         """
+        timestamp_normalized = _SecondsNanos.from_flexible(timestamp).to_nanoseconds()
+        current_time_ns = time.time_ns()
+        enqueue_dict_timestamp_diff = (current_time_ns - timestamp_normalized) / 1e9
+
         for channel, value in channel_values.items():
             self.enqueue(channel, timestamp, value)
 
+        current_time_ns = time.time_ns()
+        last_enqueue_timestamp = (current_time_ns - timestamp_normalized) / 1e9
```

Comment:
these 1e9s everywhere seem kinda messy, maybe we should just do the conversion inside `_add_metric`?

Reply:
this also isnt a timestamp but instead a delta right?